### PR TITLE
add example of using Tables.dictcolumntable

### PR DESCRIPTION
**Pull request by bkamins**
Related to https://github.com/JuliaData/DataFrames.jl/issues/3382

### CODE REVIEW

**Comment by bkamins** [comment]
@nalimilan - can we merge this and in https://github.com/JuliaData/DataFrames.jl/issues/3382 continue the discussion what we should change? (I think even if we make the change it is useful to give this tip to users)

File: docs/src/man/basics.md [hunk 1]
```diff
@@ -175,6 +175,40 @@ julia> DataFrame([(a=1, b=0), (a=2, b=0)])
...
+julia> DataFrame(Tables.dictcolumntable(source))
+2×3 DataFrame
+ Row │ type    radius   side
+     │ String  Int64?   Int64?
+─────┼──────────────────────────
+   1 │ circle       10  missing
+   2 │ square  missing       20
+```
+
+The `Tables.dictcolumntable` role is to make sure that the `DataFrame` constructor gets information
```
Comment (nalimilan) [inline docs/src/man/basics.md:202]:
```suggestion
The role of `Tables.dictcolumntable` is to make sure that the `DataFrame` constructor gets information
```

**Review by nalimilan — APPROVED** [review]

File: docs/src/man/basics.md [hunk 2]
```diff
@@ -175,6 +175,40 @@ julia> DataFrame([(a=1, b=0), (a=2, b=0)])
...
+
+```
+julia> source = [(type="circle", radius=10), (type="square", side=20)]
+2-element Vector{NamedTuple{names, Tuple{String, Int64}} where names}:
+ (type = "circle", radius = 10)
+ (type = "square", side = 20)
+```
+
+If you want to create a data frame from such data containing all columns present in at least
+one of the source observations, and holding `missing` entry if some column is not present then
```
Comment (quinnj) [inline docs/src/man/basics.md:189]:
```suggestion
one of the source observations, with a `missing` entry if some column is not present then
```

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**